acceptancetests/assess_network_spaces.py

```diff
@@ -149,18 +148,24 @@ def assert_machine_connectivity(self, client):
         machines = yaml.safe_load(
             client.get_juju_output(
                 'list-machines', '--format=yaml'))['machines']
+        spaces = non_infan_subnets(
```

Comment:
I would make these nicer get_spaces, get_machines functions since it's a common action you will take in the assert functions. If you don't want to pass in the machines / spaces, then make these yaml loads a function. 

I think I would change the testing_iterations call so that you called each test like 

```
        for test in alltests:
            try:
                test(client, spaces, machines)
```

Reply:
In general, keeping your assert methods short and easy to read is ideal. This brings you closer to avoiding complexity in those methods.

Reply:
For example, assert_internet_connection could look something like this. Move out the logic into assess methods, and leave those in the assert so you can read it easily and understand your assertions.

```
    def assert_internet_connection(self, client):
        """Test that targets can ping their default route.

        :param client: Juju client
        """
        log.info('Assessing internet connection.')
        for unit in client.get_status().iter_machines(containers=False):
            log.info("Assessing internet connection for "
                     "machine: {}".format(unit[0]))
            assess_internet_routing(client)
            assses_default_route(client)
        log.info('PASSED')
```
```
           def assess_internet_routing()
            try:
                routes = client.run(['ip route show'], machines=[unit[0]])
            except subprocess.CalledProcessError:
                raise TestFailure(
                        'Could not connect to address for unit: {0}, '
                        'unable to find default route.'.format(unit[0]))
```
```
             def assses_default_route()
            default_route = re.search(r'(default via )+([\d\.]+)\s+',
                                      json.dumps(routes[0]))
            if not default_route:
                raise TestFailure(
                        'Default route not found for {}'.format(unit[0]))
```

Reply:
Ironically, this method was lifted almost directly from the assess_network_health script. I think if I were to rewrite it, I'd simplify it to just a ping of some external IP, like 8.8.8.8.